### ISSUE TITLE
false positive for "Unnecessary semicolon", leading to syntax errors with --fix.

### ISSUE THREAD

**Issue by pygy** [palantir/tslint#2871]
### Bug Report

- __TSLint version__: 4.5.1
- __TypeScript version__: 2.3.4
- __Running TSLint via__: CLI and vsCode

#### TypeScript code being linted

```ts
//foo.ts
export function x(foo: boolean) {
    switch (foo) {
        case true: console.log(true); break
        case false: console.log(true); break
    }
}
```

with no `tslint.json` configuration

#### Actual behavior

```
foo.ts[3, 37]: Unnecessary semicolon
foo.ts[4, 38]: Unnecessary semicolon
```
Those are auto-corrected to the following, which is a syntax error:

```ts
export function x(foo: boolean) {
    switch (foo) {
        case true: console.log(true) break
        case false: console.log(true) break
    }
}
```
#### Expected behavior

No false positive

**Comment by adidahiya** [comment]
looks like a duplicate of https://github.com/palantir/tslint/issues/2564. can you try with tslint 5.2+?

also you should include your tslint.json in bug reports. "no `tslint.json` configuration" isn't really a valid way to run tslint...

**Comment by pygy** [comment]
Thanks for the reply, and oohh, my bad, it is a project I took over and I hadn't noticed that `tslint` had already been installed... plus I'm a bit tired...

Hence the old version (I had ignored the `npm install tslint` output), and my belief that it ran without a config file. For the record, here it is. Bumping to `tslint@5.2.0+` (`5.4.2`, to be exact) fixed the problem

```JSON
{
  "rules": {
    "class-name": true,
    "comment-format": [
      false,
      "check-space"
    ],
    "no-duplicate-variable": true,
    "no-eval": true,
    "no-internal-module": false,
    "no-trailing-whitespace": true,
    "no-unused-variable": true,
    "no-var-keyword": true,
    "one-line": [
      true,
      "check-open-brace",
      "check-whitespace"
    ],
    "quotemark": [
      false,
      "double"
    ],
    "semicolon": [true, "never"],
    "triple-equals": [
      true,
      "allow-null-check"
    ],
    "typedef-whitespace": [
      false,
      {
        "call-signature": "nospace",
        "index-signature": "nospace",
        "parameter": "nospace",
        "property-declaration": "nospace",
        "variable-declaration": "nospace"
      }
    ],
    "variable-name": [
      true,
      "ban-keywords"
    ],
    "whitespace": [
      true,
      "check-branch",
      "check-decl",
      "check-operator",
      "check-type"
    ]
  }
}
```
